### ISSUE TITLE
Docs Docs Docs

### ISSUE THREAD

**Issue by emilypi** [scalaz/scalaz#1974]
## Who's your favorite Dwarf? Clearly its

![image](https://user-images.githubusercontent.com/10459342/44959741-76366e80-aec1-11e8-8874-a00753f63cf2.png)

DOC. We need to get the documentation for Scalaz 8 up to date. Lots of new things have been added, and as with most things, documentation runs a little behind. This would be a good first issue for anyone who wants to learn more about the Scalaz 8 ecosystem and the new design choices we've made. Additionally, this would give someone the chance to play around with Tut. I am going to take part in this as well, so the chance to pair up on documentation with me is available. 

We also need general Scaladoc documentation within some parts of the code. Types are documentation in strongly typed languages, but Scala is a dynamically typed, low-level scripting language, and therefore needs our help explaining what types do what, when, where, and how.

**Comment by fommil** [comment]
But dwarf is not my favourite doc!

![](http://zerocarbonista.com/wp-content/uploads/2012/01/doc-back-to-the-future.jpg)

**Comment by abcoates** [comment]
Emily, are you still looking for help with this?  I'd be happy to help out.  Good documentation is in my own self-interest.

**Comment by emilypi** [comment]
Yes we are, @abcoates! Feel free to take a look and see what needs done. Most of the typeclasses and data are not sufficiently documented, and would benefit from your help if you choose to take on the task. It would be a fantastic learning experience as well.

**Comment by abcoates** [comment]
Indeed.  Can you tell me about Tut?  Also, do have a "hit list" of areas where documentation is weak, or is there a need just to go over everything top-to-bottom?

**Comment by abcoates** [comment]
Also, are you going to cover all documentation under this one issue, or will it make sense to break the documentation for different areas out into different issues?  I have little experience of issues in GitHub (mostly use JIRA), so I don't know what works best.

**Comment by abcoates** [comment]
@emilypi, I'm on the 'series/8.0.x' branch, and when I run 'sbt doc', it produces definitions on the 'scalaz.tests' page, but none on the 'scalaz' page.  Is there some trick to generating the scaladoc correctly, or should it really just be as simple as running 'sbt doc'?  Thanks.

**Comment by abcoates** [comment]
Anyway, I've started looking at this, let me know if you have anything else that I should know, @emilypi.  Thanks!

**Comment by abcoates** [comment]
> @emilypi, I'm on the 'series/8.0.x' branch, and when I run 'sbt doc', it produces definitions on the 'scalaz.tests' page, but none on the 'scalaz' page. Is there some trick to generating the scaladoc correctly, or should it really just be as simple as running 'sbt doc'? Thanks.

Ignore this comment, seems to be a configuration thing - it's only generating scaladoc for the tests, not the main classes.

**Comment by emilypi** [comment]
Thanks for looking into it @abcoates - I'm glad you got it working. Lets cover all the documentation under this issue. I'll go through the project and see if i can come up with an in-depth list of stuff we can tick off as we go

**Comment by abcoates** [comment]
No worries.  I'm just getting up to speed with how the project is set-up in sbt, and what consequences that has for how the Scaladoc generation works.

**Comment by abcoates** [comment]
@emilypi, is the expectation that a lot of the missing Scalaz 8 documentation will copy the existing Scalaz 7 documentation?  Is there any reason why that should be the case, for things that are (or are mostly) the same?  Thanks.

**Comment by hamednourhani** [comment]
Hi @abcoates . 
Can you share part of this task with me . I really like to be participate and learn more about Scala.

**Comment by emilypi** [comment]
@abcoates Yes. However, improve where you can if you see missing bits

**Comment by hamednourhani** [comment]
@emilypi . where can I start from. Do you have any idea?
